From 36d7a831e15ceb84e937122c87d01c14242dc377 Mon Sep 17 00:00:00 2001
From: Sang Yan <sangyan@huawei.com>
Date: Tue, 27 Jul 2021 22:02:17 +0800
Subject: [PATCH] config: enable kernel hotupgrade features by default
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 36d7a831e15ceb84e937122c87d01c14242dc377


hulk inclusion
category: feature
bugzilla: 48159
CVE: N/A

------------------------------

enable kernel hot upgrade features by default:
1 add pin mem method for checkpoint and restore:
 CONFIG_PIN_MEMORY=y
 CONFIG_PIN_MEMORY_DEV=m
2 add pid reserve method for checkpoint and restore
 CONFIG_PID_RESERVE=y
3 add cpu park method
 CONFIG_ARM64_CPU_PARK=y
4 add quick kexec support for kernel
 CONFIG_QUICK_KEXEC=y
5 add legacy pmem support for arm64
 CONFIG_ARM64_PMEM_RESERVE=y
 CONFIG_ARM64_PMEM_LEGACY_DEVICE=y
 CONFIG_PMEM_LEGACY=m

Signed-off-by: Sang Yan <sangyan@huawei.com>
Signed-off-by: Jingxian He <hejingxian@huawei.com>
Signed-off-by: Zhu Ling <zhuling8@huawei.com>
Reviewed-by: Chen Wandun <chenwandun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/configs/openeuler_defconfig | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/configs/openeuler_defconfig b/arch/arm64/configs/openeuler_defconfig
index cb7ea0f4b282..fd62a5d7d069 100644
--- a/arch/arm64/configs/openeuler_defconfig
+++ b/arch/arm64/configs/openeuler_defconfig
@@ -408,10 +408,13 @@ CONFIG_PARAVIRT_TIME_ACCOUNTING=y
 CONFIG_KEXEC=y
 # CONFIG_KEXEC_FILE is not set
 CONFIG_CRASH_DUMP=y
+CONFIG_ARM64_CPU_PARK=y
 # CONFIG_XEN is not set
 CONFIG_FORCE_MAX_ZONEORDER=14
 CONFIG_UNMAP_KERNEL_AT_EL0=y
 CONFIG_RODATA_FULL_DEFAULT_ENABLED=y
+CONFIG_ARM64_PMEM_RESERVE=y
+CONFIG_ARM64_PMEM_LEGACY_DEVICE=y
 # CONFIG_ARM64_SW_TTBR0_PAN is not set
 CONFIG_ARM64_TAGGED_ADDR_ABI=y
 CONFIG_ARM64_ILP32=y
@@ -711,6 +714,7 @@ CONFIG_CRYPTO_AES_ARM64_BS=m
 #
 CONFIG_CRASH_CORE=y
 CONFIG_KEXEC_CORE=y
+CONFIG_QUICK_KEXEC=y
 CONFIG_SET_FS=y
 CONFIG_KPROBES=y
 CONFIG_JUMP_LABEL=y
@@ -1041,6 +1045,8 @@ CONFIG_FRAME_VECTOR=y
 # CONFIG_GUP_BENCHMARK is not set
 # CONFIG_READ_ONLY_THP_FOR_FS is not set
 CONFIG_ARCH_HAS_PTE_SPECIAL=y
+CONFIG_PIN_MEMORY=y
+CONFIG_PID_RESERVE=y
 # end of Memory Management options
 
 CONFIG_NET=y
@@ -3285,6 +3291,7 @@ CONFIG_TCG_TIS_ST33ZP24=y
 CONFIG_TCG_TIS_ST33ZP24_I2C=y
 CONFIG_TCG_TIS_ST33ZP24_SPI=y
 # CONFIG_XILLYBUS is not set
+CONFIG_PIN_MEMORY_DEV=m
 # end of Character devices
 
 # CONFIG_RANDOM_TRUST_CPU is not set
@@ -6007,6 +6014,7 @@ CONFIG_ND_BTT=m
 CONFIG_BTT=y
 CONFIG_OF_PMEM=m
 CONFIG_NVDIMM_KEYS=y
+CONFIG_PMEM_LEGACY=m
 CONFIG_DAX_DRIVER=y
 CONFIG_DAX=y
 CONFIG_DEV_DAX=m
-- 
2.26.2

